--- crosstool-ng-1.18.0/scripts/build/libc/newlib.sh.ORIG	2013-03-04 13:38:22.104481218 -0500
+++ crosstool-ng-1.18.0/scripts/build/libc/newlib.sh	2013-03-04 13:38:57.464076737 -0500
@@ -9,7 +9,7 @@
     local libc_src
     local avr32headers_src
 
-    libc_src="ftp://sources.redhat.com/pub/newlib"
+    libc_src="ftp://sourceware.org/pub/newlib"
     avr32headers_src="http://dev.doredevelopment.dk/avr32-toolchain/sources"
 
     if [ "${CT_LIBC_NEWLIB_CUSTOM}" = "y" ]; then
